src/Operation/DropSearchIndex.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/*
+ * Copyright 2015-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Operation;
+
+use MongoDB\Driver\Command;
+use MongoDB\Driver\Exception\RuntimeException as DriverRuntimeException;
+use MongoDB\Driver\Server;
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\Exception\UnsupportedException;
+
+use function current;
+
+/**
+ * Operation for the dropSearchIndexes command.
+ *
+ * @see \MongoDB\Collection::dropSearchIndexes()
+ * @see https://mongodb.com/docs/manual/reference/command/dropSearchIndexes/
+ */
+class DropSearchIndex implements Executable
+{
+    /** @var string */
+    private $databaseName;
+
+    /** @var string */
+    private $collectionName;
+
+    /** @var string */
+    private $indexName;
+
+    /**
+     * Constructs a dropSearchIndex command.
+     *
+     * @param string $databaseName   Database name
+     * @param string $collectionName Collection name
+     * @param string $indexName      Index name
+     * @throws InvalidArgumentException for parameter parsing errors
+     */
+    public function __construct(string $databaseName, string $collectionName, string $indexName)
+    {
+        if ($indexName === '') {
+            throw new InvalidArgumentException('$indexName cannot be empty');
+        }
+
+        $this->databaseName = $databaseName;
+        $this->collectionName = $collectionName;
+        $this->indexName = $indexName;
+    }
+
+    /**
+     * Execute the operation.
+     *
+     * @see Executable::execute()
+     * @return array|object Command result document
+     * @throws UnsupportedException if write concern is used and unsupported
+     * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
+     */
+    public function execute(Server $server)
+    {
+        $cursor = $server->executeCommand($this->databaseName, $this->createCommand());
+
+        return current($cursor->toArray());
```

Comment:
Offhand, I don't recall if [`dropIndexes`https://www.mongodb.com/docs/manual/reference/command/dropIndexes/] returns any special fields, but the cluster/operation time could be relevant for folks using change streams or [causal consistency](https://github.com/mongodb/specifications/blob/master/source/causal-consistency/causal-consistency.rst). Emphasis on _could_ here, as I doubt this is used very much at all in practice.

This would mainly be for API consistency, as I don't believe we have any operation helpers in the core classes (e.g. Collection) with a `void` return type. A separate concern might be that we wouldn't be able to change the signature later if needed (IIRC this conflicts with overrides in child classes).

Note: an annoyance here is that DropIndexes supports a `typeMap` option purely to provide control the returned command result. I suppose we'd also support that here for consistency.

Reply:
> the cluster/operation time could be relevant for folks using change streams or [causal consistency](https://github.com/mongodb/specifications/blob/master/source/causal-consistency/causal-consistency.rst).

I think monitoring features of PHPC a made made for this. I really don't see the point.